### ISSUE TITLE
Midi Playback Support

### ISSUE THREAD

**Issue by luciusDXL** [luciusDXL/XL-Engine#19]
Obviously the engine needs to support Midi playback. Initially the engine will use software synthesis - adding hardware support later (that will be another ticket).

**Comment by luciusDXL** [comment]
This has been partially implemented by integrating Wild Midi. However, while the system can play back midi, more works needs to be done before release - \* integrate fluidsynth (for sf2 support), \* proper threading, \* better sound management (currently is very bare bones).

**Comment by luciusDXL** [comment]
The threading and improved sound support issues are done. All that is remaining in this task are: fluidsynth integration and game specific requirements.

**Comment by psi29a** [comment]
For those following along, an OPL3 (Sound Blaster 16) soundfont that can be used (CC-BY-SA) is here:
http://zandro.freeunixhost.com/opl3/

You can use unsf  to extract the soundfont to gus-patches.
https://launchpad.net/~tigerfishdaisy/+archive/ubuntu/ppa-fish/
http://alsa.opensrc.org/Unsf

**Comment by luciusDXL** [comment]
Music playback is essentially complete, barring any game specific functionality I have to add. The XL Engine can now play midi using GUS patches using Wild Midi, Sound Fonts (sf2) using FluidSynth and play back OGG files instead using libvorbis.
